src/LendController.sol

```diff
@@ -0,0 +1,156 @@
+        loanFacilitator = _facilitator;
+    }
+
+    event SetTermsApproval(address indexed from, address indexed termsContract, bool approved);
+
+    function setTermsApproval(address termsContract, bool approved) external {
+        termsApprovals[msg.sender][termsContract] = approved;
+
+        emit SetTermsApproval(msg.sender, termsContract, approved);
+    }
+
+    function purchaseNFT() external {
+        // allows anyone to call in and seize a seize NFT that
+        // the controller holds the lend ticket for and pays the lenders
+        // purchase price must be >= totalOwed on loan
+    }
+
+    function liquidateNFT(uint256 loanId) external {
+        // TBD if controller will have one liquidation mechanism 
+        // or will call out to the terms contract for implement
+        // update lenderBalance based on lenderLoanShares * sale value 
+    }
+
+    // used to lend to an exisiting loan
+    function lend(
+        uint256 loanId,
+        uint256 tokenId,
+        address nft,
+        BorrowRequest memory request,
+        bytes calldata data
+    ) external {
+        _lend(loanId, tokenId, nft, request, false, data);
+    }
+
+    function _lend(
+        uint256 loanId,
+        uint256 tokenId,
+        address nft,
+        BorrowRequest memory request,
+        bool skipIsBuyoutCheck,
+        bytes calldata data
+    ) internal {
+        LoanTerms memory terms = NFTLoanTermsSource(request.termsSource).terms(tokenId, msg.sender, request.terms.loanAsset, '');
+
+        ERC20 loanAsset = ERC20(request.terms.loanAsset);
+        uint256 lenderTotal;
+
+        for(uint i = 0; i < request.lendersSpecs.length; i++) {
+            LenderSpec memory info = request.lendersSpecs[i];
+            require(termsApprovals[info.lender][request.termsSource], 'lender has not approved terms source');
+
+            if(info.pull) {
+                ERC20(request.terms.loanAsset).safeTransferFrom(info.lender, address(this), info.amount);
+            } else {
+                lenderBalance[info.lender][address(loanAsset)] -= info.amount;
+            }
+            lenderTotal += info.amount;
+
+            lenderLoanShares[loanId][info.lender] = info.amount * ONE / terms.amount;
+        }
+
+        loanAsset.approve(address(loanFacilitator), type(uint256).max);
+
+        if(skipIsBuyoutCheck){
+            require(lenderTotal == terms.amount);
+        } else {
+            (,,, uint40 lastAccumulatedTimestamp,,,,,,,) = loanFacilitator.loanInfo(loanId);
+            
+            if(lastAccumulatedTimestamp != 0) {
+                require(terms.allowBuyouts);
+                require(lenderTotal == terms.amount + loanFacilitator.interestOwed(loanId));
+            } else {
+                require(lenderTotal == terms.amount);
+            }
+        }
+
+        loanFacilitator.lend(
+            loanId,
+            terms.interestRate,
+            terms.amount,
+            terms.durationSeconds,
+            address(this)
```

Comment:
this actually isn't quite right: we do not want to transfer the lend ticket to this contract. Because Backed has eager transfers for loan repayments, and we need to track whose funds are whose in the LendController, we will need to deploy a contract solely for the purpose of holding the lend ticket and receiving repayment. We will need one such contract for every lend ticket

Reply:
Non-eager payouts are better for composability but worse for gas in the EOA case: its into lender returns to have to come back and withdraw payback

Reply:
with regards to collateral seizure, are any of the lenders able to call seizeCollateral? does this newly deployed contract need to then know about liquidation strategy?

Reply:
I think the newly deployed contract would just allow the parent to make seize calls and send the collateral to itself/somewhere. parent (i.e. LendController) would control auction. 

And yeah I think possibly anyone could call seizeCollateral, so long as what happens next (e.g. what kind of auction) is pre-agreed on